<!--
@author:    Johann Schmidt
@date:      November 2019
@brief:     Script used to spawn a turtlebot.
@todo:
-->

<launch>
    <arg name="model" default="$(env TURTLEBOT3_MODEL)" doc="model type [burger, waffle, waffle_pi]"/>
    <arg name="name"  default="tb3_0"/>
    <arg name="pos_x" default="-1.0"/>
    <arg name="pos_y" default="-.5"/>
    <arg name="pos_z" default=" 0.0"/>
    <arg name="yaw"   default=" 0.0"/>

    <group ns = "$(arg name)">
    <param name="robot_description" command="$(find xacro)/xacro --inorder $(find turtlebot3_description)/urdf/turtlebot3_$(arg model).urdf.xacro" />

    <node pkg="robot_state_publisher" type="robot_state_publisher" name="robot_state_publisher" output="screen">
      <param name="publish_frequency" type="double" value="50.0" />
      <param name="tf_prefix" value="$(arg name)" />
    </node>

    <node name="spawn_urdf" pkg="gazebo_ros" type="spawn_model" args="-urdf -model $(arg name) -x $(arg pos_x) -y $(arg pos_y) -z $(arg pos_z) -Y $(arg yaw) -param robot_description" />
  </group>
</launch>